scripts/test-rpc.sh

```
@@ -0,0 +1,614 @@
+
+    ensure_category_header "$category"
+
+    # Add curl timing flags: -w outputs timing info, -o /dev/stderr sends response to stderr
+    # This allows us to capture timing separately from response
+    local timed_curl="${curl_cmd% \'*\'} -w '\n__TIMING__{\"total\":%{time_total},\"dns\":%{time_namelookup},\"connect\":%{time_connect},\"ttfb\":%{time_starttransfer}}__TIMING__' -s"
+
+    local full_output
+    local exit_code
+    # Cross-platform timing fallback (milliseconds)
+    # macOS date doesn't support %N, so we use a simpler approach
+    local start_time=$(python3 -c "import time; print(int(time.time() * 1000))" 2>/dev/null || echo "0")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)